python/bind_slice_methods.h

```diff
@@ -133,8 +164,12 @@ void bind_slice_methods(pybind11::class_<T, Ignored...> &c) {
   }
   if constexpr (std::is_same_v<T, DataArray> ||
                 std::is_same_v<T, DataArrayView>) {
+    c.def("__getitem__", &slicer<T>::get_by_value, py::keep_alive<0, 1>());
     c.def("__setitem__", &slicer<T>::template set<DataArrayView>);
     c.def("__setitem__", &slicer<T>::template set_range<DataArrayView>);
+    c.def("__setitem__", &slicer<T>::template set_by_value<Variable>);
+    c.def("__setitem__", &slicer<T>::template set_by_value<VariableView>);
```

Comment:
`Variable` is implicitly convertible to views, so the first overload should be redundant.

Reply:
Thanks, I have also removed the Variable templates from elsewhere in the file where a VariableView template also exists. Let me know if there is something I have missed and the others are needed.